### PR TITLE
Set up screenshot comparisons using reg-suit

### PR DESCRIPTION
**Pull request by janpaul123**
## Summary

Set up screenshot comparison tools. On every PR, we detect visual differences in stories and post them to the PR message.

## Test plan

This adds tests.

## Versioning impact

No impact.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -124,9 +128,10 @@
...
-    "screenshot-debug": "npm run screenshot -- --parallel 1 --debug --puppeteer-launch-config '{\"headless\": false}'",
-    "ci": "npm run install-ci && NODE_ENV=production npm run build && npm run lint && npm run flow && npm test && NODE_ENV=production npm run screenshot-ci"
+    "screenshot-ci": "npm run screenshot -- --puppeteer-launch-config '{\"executablePath\": \"/usr/bin/google-chrome\", \"headless\": false, \"args\":[\"--use-gl=swiftshader\", \"--no-sandbox\",\"--disable-setuid-sandbox\", \"--disable-dev-shm-usage\", \"--headless\", \"--mute-audio\", \"--user-agent=ChromeScreenshotTesting\"]}'",
+    "screenshot-debug": "npm run screenshot -- --parallel 1 --debug --puppeteer-launch-config '{\"headless\": false, \"--args\":[\"--user-agent=ChromeScreenshotTesting\"]}'",
+    "ci-commands-for-cruise-proprietary-repo": "npm run install-ci && NODE_ENV=production npm run build && npm run lint && npm run flow && npm test && NODE_ENV=production npm run screenshot-ci",
```
Comment (janpaul123) [inline package.json]: 👍 neat.

File: package.json [hunk 2]
```diff
@@ -124,9 +128,10 @@
...
-    "screenshot-ci": "npm run screenshot -- --puppeteer-launch-config '{\"executablePath\": \"/usr/bin/google-chrome\", \"headless\": false, \"args\":[\"--use-gl=swiftshader\", \"--no-sandbox\",\"--disable-setuid-sandbox\", \"--disable-dev-shm-usage\", \"--headless\", \"--mute-audio\"]}'",
-    "screenshot-debug": "npm run screenshot -- --parallel 1 --debug --puppeteer-launch-config '{\"headless\": false}'",
-    "ci": "npm run install-ci && NODE_ENV=production npm run build && npm run lint && npm run flow && npm test && NODE_ENV=production npm run screenshot-ci"
+    "screenshot-ci": "npm run screenshot -- --puppeteer-launch-config '{\"executablePath\": \"/usr/bin/google-chrome\", \"headless\": false, \"args\":[\"--use-gl=swiftshader\", \"--no-sandbox\",\"--disable-setuid-sandbox\", \"--disable-dev-shm-usage\", \"--headless\", \"--mute-audio\", \"--user-agent=ChromeScreenshotTesting\"]}'",
+    "screenshot-debug": "npm run screenshot -- --parallel 1 --debug --puppeteer-launch-config '{\"headless\": false, \"--args\":[\"--user-agent=ChromeScreenshotTesting\"]}'",
```
Comment (janpaul123) [inline package.json]: These user-agent changes would be great to also make to the internal repo!

File: docs/src/jsx/utils/codeSandboxUtils.js [hunk 3]
```diff
@@ -7,5 +7,6 @@
 //  You may not use this file except in compliance with the License.
 
 export function inScreenshotTests() {
-  return navigator.userAgent.includes("HeadlessChrome");
+  // screenshot tests are not always in a headless Chrome, so need to check for "selectedKind" in the URL as well.
+  return navigator.userAgent.includes("HeadlessChrome") || window.location.search.includes("selectedKind");
```
Comment (jtbandes) [inline docs/src/jsx/utils/codeSandboxUtils.js]: nit: URLSearchParams(window.location.search).has("selectedKind")

File: package.json [hunk 4]
```diff
@@ -77,6 +77,10 @@
     "react-svg-loader": "^2.1.0",
     "react-virtualized-select": "^3.1.3",
     "redux-devtools-extension": "^2.13.7",
+    "reg-notify-github-plugin": "^0.7.19",
```
Comment (jtbandes) [inline package.json]: maybe pin these exactly? could help with security vulnerabilities in the future?

File: packages/regl-worldview/src/stories/cameraState.stories.js [hunk 5]
```diff
@@ -148,7 +148,7 @@ class CameraStateStory extends React.Component<Props, State> {
   }
 }
 
-const stories = storiesOf("Worldview", module);
+const stories = storiesOf("Worldview - cameraState", module);
```
Comment (jtbandes) [inline packages/regl-worldview/src/stories/cameraState.stories.js]: Looks like you can use `/` for nesting: https://storybook.js.org/docs/basics/writing-stories/#nesting-stories

File: packages/regl-worldview/src/stories/Cubes.stories.js [hunk 6]
```diff
@@ -82,7 +83,8 @@ class DynamicCubes extends React.Component<any, any> {
   }
 }
 
-storiesOf("Worldview", module)
+storiesOf("Worldview - Cubes", module)
```
Comment (jtbandes) [inline packages/regl-worldview/src/stories/Cubes.stories.js]: nit: consistent capitalization with other stories would be nice

File: docs/src/jsx/utils/codeSandboxUtils.js (hunk 3 above)
Comment (davidswinegar) [reply docs/src/jsx/utils/codeSandboxUtils.js]: Updated this to match other use agent tests

**Pull request merged**